### PR TITLE
Razor Components Forms and Validation topic

### LINKED ISSUE #11715: Component validation scenarios topic
* Validation
* EditForm
* InputText

Add explicit churn note to topic.

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #11715 

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/razor-components/forms-validation?view=aspnetcore-3.0&branch=pr-en-us-11780)

* The starship example, which is rich and full featured, is probably a bit too much&dagger; to throw in the reader's face when introducing this content. I start with a simpler example to orient the reader to the basics of an `<EditForm>` before going into the larger, more complicated example.

  *bit too much*&dagger; This is only going to get worse later because there will be *more goodies* to show in the example. As the starship example expands, it will make an even worse *First Contact* with the Borg in this topic! :smile: lol

* Withholding this one from the sample app right now: It will :boom: in a giant merge conflict for one thing. The other consideration is the rapid churn in this area. It's best to wait until RC to roll this into the sample.

* I'm not including a screenshot here, but we certainly can if you feel that it's important. I have the copyright in the form now, so we can use a real Star Trek ship example in a screenshot. Let me know.

### CODE REVIEW

File: aspnetcore/razor-components/forms-validation.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    [Range(typeof(bool), "true", "true", 
+        ErrorMessage = "Form disallowed for unapproved ships.")]
+    public bool IsValidatedDesign { get; set; }
+
+    [Required]
+    public DateTime ProductionDate { get; set; }
+}
+```
+
+The following Starship Database form validates user input using the validation defined in the `Starship` model:
```
Comment (Rick-Anderson) [inline aspnetcore/razor-components/forms-validation.md]: The following form validates user input using the validation defined in the `Starship` model:

File: aspnetcore/razor-components/forms-validation.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    private Starship starship = new Starship();
+
+    void HandleValidSubmit()
+    {
+        Console.WriteLine("OnValidSubmit");
+    }
+}
+```
+
+The `<EditForm>` creates an `EditContext` as a [cascading value](xref:razor-components/components#cascading-values-and-parameters) that tracks metadata about the edit process, including what fields have been modified and the current validation messages. The `<EditForm>` also provides convenient events for valid and invalid submits (`OnValidSubmit`, `OnInvalidSubmit`). Alternatively, you can use `OnSubmit` directly if you want to trigger the validation and check field values yourself.
```
Comment (Rick-Anderson) [inline aspnetcore/razor-components/forms-validation.md]: Alternatively, use `OnSubmit` to trigger the validation and check field values with user supplied validation code.

Or something like that.

File: aspnetcore/razor-components/forms-validation.md (hunk 1 above)
Comment (guardrex) [reply aspnetcore/razor-components/forms-validation.md]: The `Starship` model :smile:. HA! Lov'in it! :heart:

File: aspnetcore/razor-components/forms-validation.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+The following `ExampleModel` type defines validation logic using data annotations:
+
+```csharp
+using System.ComponentModel.DataAnnotations;
+
+public class ExampleModel
+{
+    [Required]
+    [StringLength(10, ErrorMessage = "Name is too long.")]
```
Comment (seriouz) [inline aspnetcore/razor-components/forms-validation.md]: Please explain how localization with client side Blazor or a shared lib could be done.

Comment (guardrex) [reply aspnetcore/razor-components/forms-validation.md]: @seriouz This PR has already merged. It's best to open new issues using the **This Page** feedback button at the bottom of the topic. No worries on your question tho ... I can answer that here.

*It's in progress now.* See :point_right: https://github.com/aspnet/AspNetCore/issues/9386 ... It's due (tentatively) for Preview 8. We'll get it documented for the Preview 8 release. In the meantime, if you have design comments or questions, you can ask the engineers on that issue.

**Pull request merged**